src/Sulu/Bundle/MarkupBundle/Markup/LinkTag.php

```diff
@@ -209,21 +214,39 @@ private function getContent(array $attributes)
     /**
      * @param mixed $href
      *
-     * @return array{uuid: string|null, anchor: string|null}
+     * @return array{uuid: string|null, query: string|null, anchor: string|null}
      */
-    private function getUuidAndAnchorFromHref($href): array
+    private function getPartsFromHref($href): array
     {
         $href = (string) $href ?: null;
 
-        /** @var string[] $hrefParts */
-        $hrefParts = $href ? \explode('#', $href, 2) : [];
-
-        $uuid = $hrefParts[0] ?? null;
-        $anchor = $hrefParts[1] ?? null;
+        $query = null;
+        $anchor = null;
+
+        $hasQuery = strpos($href, '?');
```

Comment:
@exastion  Something like `my.url/#hello?page=1` means that `hello?page=1` is the value of the anchor and it still a valid anchor. This is as example used in Single Page Application and are valid anchors. Sulu example use this kind of special anchor for frontend navigation.

That's why its important first to split the string by `#` and then go with split the first part by `?` and not the other way around. So there is no problem also from backward compatibility we are not allowed to change to seperate attributes and as said its not really required.

So the following examples should show you best how the end result should look like:

```bash
<uuid>?test=query#anchor

ID: <uuid>
query: test=query
anchor: anchor
```


```bash
<uuid>#anchor?test=query

ID: <uuid>
query: 
anchor: anchor?test=query
```

```bash
<uuid>?test=query

ID: <uuid>
query: test=query
anchor: 
```

See https://3v4l.org/j94IN#v7.2.34 how to correctly split the string. A `?` is a valid symbol as part of the anchor so we first need to split by `#` and then split the part before the anchor by `#`.

Reply:
I did not know about these specifics up to now, I even thought in the beginning, query and anchor would be the other way around.. Well you always learn :)
This made things also easier.

Reply:
Also nice to know the `#anchor` is never send to a server its only available for the browser. So if your browser requests `https://sulu.io/services/support#chat` a request is only send to `https://sulu.io/services/support` and browsers is handling `#chat` to scroll to the correct part. Or a javascript library is handling the `#anchor` like its done in the sulu admin.